backup-n-restore/hack/common.sh

```diff
@@ -187,3 +187,196 @@ restore_finished() {
+
+       rm -rf $managedclusterkubeconfig
+       if [ $? -eq 0 ]
+       then
+           echo_green "tmp kubeconfig $managedclusterkubeconfig deleted"
+       fi
+       
+       rm -rf $newbootstraphubkubeconfig
+       if [ $? -eq 0 ]
+       then
+	   echo_green "tmp kubeconfig $newbootstraphubkubeconfig deleted"
+       fi
+
+       cat << EOF | oc  apply -f -
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: system:open-cluster-management:managedcluster:bootstrap:${managedclustername}
+rules:
+- apiGroups:
+  - certificates.k8s.io
+  resources:
+  - certificatesigningrequests
+  verbs:
+  - create
+  - get
+  - list
+  - watch
+- apiGroups:
+  - cluster.open-cluster-management.io
+  resources:
+  - managedclusters
+  verbs:
+  - get
+  - create
+EOF
+
+       cat << EOF | oc  apply -f -
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  managedFields:
+  name: system:open-cluster-management:managedcluster:bootstrap:${managedclustername}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: system:open-cluster-management:managedcluster:bootstrap:${managedclustername}
+subjects:
+- kind: ServiceAccount
+  name: ${managedclustername}-bootstrap-sa
+  namespace: ${managedclustername}
+EOF
+
+        cat << EOF | oc  apply -f -
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: open-cluster-management:managedcluster:${managedclustername}
+rules:
+- apiGroups:
+  - certificates.k8s.io
+  resources:
+  - certificatesigningrequests
+  verbs:
+  - create
+  - get
+  - list
+  - watch
+- apiGroups:
+  - register.open-cluster-management.io
+  resources:
+  - managedclusters/clientcertificates
+  verbs:
+  - renew
+- apiGroups:
+  - cluster.open-cluster-management.io
+  resourceNames:
+  - ${managedclustername}
+  resources:
+  - managedclusters
+  verbs:
+  - get
+  - list
+  - update
+  - watch
+- apiGroups:
+  - cluster.open-cluster-management.io
+  resourceNames:
+  - ${managedclustername}
+  resources:
+  - managedclusters/status
+  verbs:
+  - patch
+  - update
+EOF
+
+	cat << EOF | oc  apply -f -
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: open-cluster-management:managedcluster:${managedclustername}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: open-cluster-management:managedcluster:${managedclustername}
+subjects:
+- apiGroup: rbac.authorization.k8s.io
+  kind: Group
+  name: system:open-cluster-management:${managedclustername}
+EOF
+
+    done
+
+}
+
+# accepts_managed_clusters iterates over all the managed clusters and accept the certificate
+accepts_managed_clusters() {
+    for managedclustername in $(oc get ns -lcluster.open-cluster-management.io/managedCluster -o jsonpath='{.items[*].metadata.name}');
+    do oc patch managedcluster ${managedclustername} -p='{"spec":{"hubAcceptsClient":true}}' --type=merge;
```

Comment:
Might be worth validating whether the original managedcluster on the primary hub had also been accepted. I'm wondering if there is an attack vector where an untrusted `ManagedCluster` slips into the inventory during the window where we're replicating details & gets accepted prior to the original hub accepting the cluster.

Reply:
Right. Going to do Thanks!